### PR TITLE
feat: a config for excluding files from being compile by default [APE-1326]

### PR DESCRIPTION
**Pull request by antazoey**
### What I did

Adds a config so that you can do:

```yaml
compile:
  exclude:
    - testlibs/*  # excludes contracts in `contracts/testlibs/`
    - *Mock.vy  # Exclude vyper files ending in `Mock`.
```

### Checklist

- [ ] All changes are completed
- [ ] New test cases have been added
- [ ] Documentation has been updated

### CODE REVIEW

File: src/ape_compile/__init__.py [hunk 1]
```diff
@@ -32,6 +37,10 @@ def warn_deprecate(cls, value):
 
         return None
 
+    @validator("exclude", pre=True)
+    def validate_exclude(cls, value):
+        return value or []
```
Comment (fubuloubu) [inline src/ape_compile/__init__.py:42]: Is this necessary?

**Review by fubuloubu — APPROVED** [review]

File: src/ape_compile/__init__.py (hunk 1 above)
Comment (antazoey) [reply src/ape_compile/__init__.py:42]: define necessary? it makes an empty list in the config not be `None`.. helped me when commenting stuff out without having to delete the whole config.

Comment (antazoey) [reply src/ape_compile/__init__.py:42]: i was doing this:

```
compile:
  exclude:
    # - *thing*
```

and i had the error. I think it should know i just mean empty list here and not `None`, or at least i dont see the harm in allowing this state.

Comment (fubuloubu) [reply src/ape_compile/__init__.py:42]: ahhhh

you were actively inserting a `None`

**Review by fubuloubu — DISMISSED** [review]

**Pull request merged**